### ISSUE TITLE
DELETE from internal table gives a "Database access" prio 1 error

### ISSUE THREAD

**Issue by whydrae** [SAP/code-pal-for-abap#290]
Hello colleagues,

The following line in a test method gives us a Y_CHECK_DB_ACCESS_IN_UT priority 1 error:

```DELETE lt_table INDEX lines( lt_table ).```

Local test class defined as:
```
CLASS ltc_test_class DEFINITION FOR TESTING
  DURATION SHORT RISK LEVEL HARMLESS FINAL.
```

Could you please check what's wrong?

Thanks!

**Comment by estevao-schultz-neto-SAP** [comment]
hello!!

If lt_table is not a persistent table, it shouldn't be the case. We will check that.
Besides, these changes were recently adopted for this check:

CHECK is being adapted so that it should act differently according to the RISK LEVEL classification of the test class under evaluation. So that:

	· RISK LEVEL HARMLESS or missing RISK LEVEL classification --> These DB Operation will be forbidden/reported: COMMIT, 
                                                                                                               DELETE, INSERT, MODIFY, ROLLBACK, SELECT and UPDATE 
                                                                                                                statements on persistent DDIC tables;
	· RISK LEVEL DANGEROUS/CRITICAL --> UPDATE, MODIFY DELETE, COMMIT, ROLLBACK operation(s) on persistent DDIC 
                                                                         tables will be forbidden/reported.
 
Besides, test classes having AMDP (this happens only in productive mode) are already excluded from the scope of the check (in other words, exempted) since the Check is only applicable on test-code.

Again, data access should be always mocked. For the mocking, one of these alternatives should be used:
OSQL Test Framework; and/or
CDS Test Framework; and/or
Test-Double Framework (isolating the database access in classes an mocking them via TDF).

**Comment by lucasborin** [comment]
The Check detected the `INDEX` is a persistent object:
![image](https://user-images.githubusercontent.com/5233413/103003680-f7d9bc80-450f-11eb-8e7f-242f88a343f5.png)

**Comment by lucasborin** [comment]
Fix will be available in version 1.11.0, but it is already implemented in the EHR. Feel free to test it.
